tests/client/test_base_client.py

```diff
@@ -469,7 +506,39 @@ async def test_prefect_httpx_client_raises_prefect_http_status_error(
         base_client_send.return_value = RESPONSE_400
         with pytest.raises(PrefectHTTPStatusError) as exc:
             await client.post(url="fake.url/fake/route", data={"evenmorefake": "data"})
-            assert (
-                "Response: {'extra_info': [{'message': 'a test error message'}]}"
-                in str(exc)
-            )
+        expected = "Response: {'extra_info': [{'message': 'a test error message'}]}"
+        assert expected in str(exc)
+
+    @pytest.mark.parametrize(
+        "mock_redirects",
+        [
+            ["https://example.org/redirect_301", "https://example.org/"],
+            ["https://example.org/redirect_302", "https://example.org/redirected_302"],
+            ["https://example.org/redirect_307", "https://example.org/redirected_307"],
+            ["https://example.org/redirect_308", "https://example.org/redirected_308"],
+        ],
+    )
+    async def test_prefect_httpx_client_follow_redirect_true(self, mock_redirects):
+        mock_transport = httpx.MockTransport(redirects)
+        client = PrefectHttpxClient(follow_redirects=True, transport=mock_transport)
+        original_url, redirected_url = mock_redirects
+        response = await client.get(original_url)
+        assert response.url == redirected_url
+
+    @pytest.mark.parametrize(
+        "mock_redirects",
+        [
+            ["https://example.org/redirect_301", "301 Moved Permanently"],
+            ["https://example.org/redirect_302", "302 Found"],
+            ["https://example.org/redirect_307", "307 Temporary Redirect"],
+            ["https://example.org/redirect_308", "308 Permanent Redirect"],
+        ],
+    )
+    async def test_prefect_httpx_client_follow_redirect_false(self, mock_redirects):
+        mock_transport = httpx.MockTransport(redirects)
+        client = PrefectHttpxClient(follow_redirects=False, transport=mock_transport)
+        original_url, redirect_response = mock_redirects
+        with pytest.raises(PrefectHTTPStatusError) as exc:
+            await client.get(original_url)
+        expected = f"Redirect response '{redirect_response}' for url '{original_url}'"
```

Comment:
This asserts the behaviour the user reported with respect to redirects:
- `prefect.exceptions.PrefectHTTPStatusError: Redirect response '307 Temporary Redirect' for url 'https://prefect-api.test.***/api/flows/'`

Reply:
Cool, I removed this test. I was just being extra cautious b/c I don't know this codebase and wasn't sure what the testing bar was, thanks!